Define case sensitivity of map<string, any> keys in Logs Data Model (#3889)
open-telemetry/opentelemetry-specification: specification/logs/data-model.md

#### Type `map<string, any>`

Value of type `map<string, any>` is a map of string keys to `any` values.

> Case sensitivity of keys MUST be preserved.

Reviewer: Looks good to me! This is also in line with what we have defined for common attributes:
https://github.com/open-telemetry/opentelemetry-specification/blob/a1c13d59bb7d0fb086df2b3e1eaec9df9efef6cc/specification/common/README.md#L35
I believe any deviation from that for log bodies would be pretty confusing and expect the definition you're adding here to conform with the general understanding anyway.

The keys in the map MUST be unique (duplicate keys are not allowed).
The enforcement of uniqueness may be performed in a variety of ways as it best

> fits the limitations of the particular implementation.

Reviewer: What does this mean? Is considering case-sensitivity or not up to SDKs then, or does this only allow SDKs to decide on how to enforce uniqueness (first write wins, last write wins, throwing an error, ...)?

Author: > does this only allow SDKs to decide on how to enforce uniqueness 

This is how I understand it.

This fragment is taken from https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute-collections

Reviewer: I think this could be clarified with:

```suggestion
The enforcement of uniqueness may be performed in a variety of ways and implementations
may choose how to enforce this requirement within their limitations.
```

Ideally it would be fixed in both places if there are no objections but if others think the current wording is unambiguous enough already, I'm fine with either.

Author: @arminru I will make a follow-up PR if this one is merged with your suggestion

#### Type `map<string, any>`

Value of type `map<string, any>` is a map of string keys to `any` values.

> Case sensitivity of keys MUST be preserved.

Reviewer 2: I'm not sure if the wording here is right. "Case sensitivity MUST be preserved" logically means "if something is case sensitive, it should stay case sensitive; if something is case insensitive, it should stay case insensitive". Doesn't it? :thinking: 

Perhaps a better wording would be:

```
The keys in the map are case-sensitive and MUST be unique (duplicate keys are not allowed).
```
